cmd/lifecycle/analyzer.go

```diff
@@ -25,6 +26,9 @@ type analyzeCmd struct {
 
 	//flags: paths to write data
 	analyzedPath string
+
+	//construct before dropping privileges
+	keychain authn.Keychain
```

Comment:
I'm not sure whether it's better to have the `keychain` here or as part of the `analyzeArgs` struct.
We thought about putting it as part of `analyzeCmd` to be consistent with other commands that don't have the `args` structure (such as the `creator`), but then we need to pass the `keychain` as an argument to the `analyze` function which feels a bit weird.
@ekcasey , what do you think?

Reply:
Maybe we should add an `args` struct to the other functions as well? It seems like an overkill...

Reply:
See https://github.com/buildpacks/lifecycle/pull/409/files#r540434516 for another case.

Reply:
In my mind this seems most analogous to the pre-connected `docker` client which we put on the `analyzeArgs` I think it makes sense to move it there.